presto-spi/src/main/java/com/facebook/presto/spi/type/Varchars.java

```diff
@@ -52,17 +53,45 @@ public static Slice truncateToLength(Slice slice, int maxLength)
         if (maxLength == 0) {
             return Slices.EMPTY_SLICE;
         }
+
+        return slice.slice(0, getTruncatedLength(slice, 0, slice.length(), maxLength));
+    }
+
+    public static int getTruncatedLength(Slice slice, int offset, int length, int codePointCount)
```

Comment:
`lengthAfterTruncation`? The 3 new functions you are adding should have consistent naming.

Reply:
`byteCount()`; add comments to explain